### PR TITLE
Add the workaround to all partial specializations of function pointers and member function pointers

### PR DESCRIPTION
**Pull request by uyha**
Problem:
- The workaround introduced by PR #390 was not applied to all specializations

Solution:
- Add the workaround to all the specializations.

### CODE REVIEW

File: include/boost/sml/aux_/utility.hpp [hunk 1]
```diff
@@ -369,9 +369,9 @@ const char *get_type_name() {
   return detail::get_type_name<T, 39>(__FUNCSIG__, make_index_sequence<sizeof(__FUNCSIG__) - 39 - 8>{});
 #elif defined(__clang__)  // __pph__
   return detail::get_type_name<T, 63>(__PRETTY_FUNCTION__, make_index_sequence<sizeof(__PRETTY_FUNCTION__) - 63 - 2>{});
-#elif defined(__GNUC__)   // __pph__
+#elif defined(__GNUC__)  // __pph__
   return detail::get_type_name<T, 68>(__PRETTY_FUNCTION__, make_index_sequence<sizeof(__PRETTY_FUNCTION__) - 68 - 2>{});
-#endif                    // __pph__
```
Comment (krzysztof-jusiak) [inline include/boost/sml/aux_/utility.hpp:374]: needs `make style` to be run

File: include/boost/sml/aux_/utility.hpp [hunk 2]
```diff
@@ -309,26 +309,26 @@ struct zero_wrapper<R (*)(TArgs...), T> {
...
   explicit zero_wrapper(R (TBase::*ptr)(TArgs...) noexcept) : ptr{ptr} {}
   auto operator()(TBase &self, TArgs... args) { return (self.*ptr)(args...); }
 
  private:
   R (TBase::*ptr)(TArgs...) noexcept {};
 };
 
-template <class R, class TBase, class... TArgs>
-struct zero_wrapper<R (TBase::*)(TArgs...) const noexcept> {
+template <class R, class TBase, class... TArgs, class T>
```
Comment (krzysztof-jusiak) [inline include/boost/sml/aux_/utility.hpp:321]: I think we should introduce tests for `wrap` to ensure the quality of the feature

File: test/ft/wrap.cpp [hunk 3]
```diff
@@ -0,0 +1,83 @@
+//
```
Comment (krzysztof-jusiak) [inline test/ft/wrap.cpp:1]: Can we just add copyright/license stuff to this file, please, it's a requirement for BOOST license for all files to have it

File: test/ft/wrap.cpp [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  expect(state_machine.is("s1"_s));
+  state_machine.process_event(next{});
+  expect(state_machine.is("s2"_s));
+  expect(i == 1);
+  state_machine.process_event(next{});
+  expect(state_machine.is(X));
+  expect(i == 2);
+};
+
+test wrap_member_functions = [] {
```
Comment (krzysztof-jusiak) [inline test/ft/wrap.cpp:42]: thanks :+1:

File: test/ft/wrap.cpp (hunk 3 above)
Comment (uyha) [reply test/ft/wrap.cpp:1]: I added the license to my test, but the test `actions_process_n_defer.cpp` doesn't have it, so I think you should check it out.

**Pull request merged**